src/Framework/NativeMethods.cs

```diff
@@ -663,6 +681,63 @@ private static LongPathsStatus IsLongPathsEnabledRegistry()
         }
     }
 
+    internal static DevDriveStatus IsOnDevDrive()
+    {
+        if (!IsWindows)
+        {
+            return DevDriveStatus.NotApplicable;
+        }
+
+        try
+        {
+            return IsOnDevDriveInternal();
+        }
+        catch
+        {
+            return DevDriveStatus.NotApplicable;
+        }
+    }
+
+    [SupportedOSPlatform("windows")]
+    private static DevDriveStatus IsOnDevDriveInternal()
+    {
+        string currentDirectory = Environment.CurrentDirectory;
+        string rootPath = Path.GetPathRoot(currentDirectory);
+        
+        if (string.IsNullOrEmpty(rootPath))
+        {
+            return DevDriveStatus.NotApplicable;
+        }
+
+        char[] volumeNameBuffer = new char[256];
+        char[] fileSystemNameBuffer = new char[256];
+        
+        bool result = GetVolumeInformation(
+            rootPath,
+            volumeNameBuffer,
+            volumeNameBuffer.Length,
+            out uint volumeSerialNumber,
+            out uint maximumComponentLength,
+            out uint fileSystemFlags,
+            fileSystemNameBuffer,
+            fileSystemNameBuffer.Length);
+
+        if (!result)
+        {
+            return DevDriveStatus.NotApplicable;
+        }
+
+        string fileSystemName = new string(fileSystemNameBuffer).TrimEnd('\0');
+        
+        // Dev Drive uses the ReFS file system
```

Comment:
Dev Drive is on ReFS but does other things, so it'd be good to be able to distinguish between just ReFS and Dev Drive.

Reply:
The new assignment is to simplify to fewer lines of code and just log on windows a message whether we're building on NTFS or ReFS. The path that we are asking about is the entrypoint project file/solution file.